app/static/app/js/components/Map.jsx

```diff
@@ -1,150 +1,179 @@
-        maxZoom: maxzoom,
-        tms: scheme === 'tms'
-      });
-
-      layers.push(this.imageryLayer);
-    }
-
-    this.leaflet = Leaflet.map(this.container, {
+    this.map = Leaflet.map(this.container, {
       scrollWheelZoom: true,
-      layers
+      measureControl: true,
+      positionControl: true
     });
 
     if (showBackground) {
-      const basemaps = [
-        L.tileLayer('//{s}.google.com/vt/lyrs=s,h&x={x}&y={y}&z={z}', {
+      this.basemaps = {
+        "Google Maps Hybrid": L.tileLayer('//{s}.google.com/vt/lyrs=s,h&x={x}&y={y}&z={z}', {
             attribution: 'Map data: &copy; Google Maps',
             subdomains: ['mt0','mt1','mt2','mt3'],
             maxZoom: 22,
             minZoom: 0,
             label: 'Google Maps Hybrid'
-        }),
-        L.tileLayer('//server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}', {
+        }).addTo(this.map),
+        "ESRI Satellite": L.tileLayer('//server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}', {
             attribution: 'Tiles &copy; Esri &mdash; Source: Esri, i-cubed, USDA, USGS, AEX, GeoEye, Getmapping, Aerogrid, IGN, IGP, UPR-EGP, and the GIS User Community',
             maxZoom: 22,
             minZoom: 0,
             label: 'ESRI Satellite'  // optional label used for tooltip
         }),
-        L.tileLayer('//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+        "OSM Mapnik": L.tileLayer('//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
             attribution: '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>',
             maxZoom: 22,
             minZoom: 0,
             label: 'OSM Mapnik'  // optional label used for tooltip
         })
-      ];
-
-      this.leaflet.addControl(Leaflet.control.basemaps({
-          basemaps: basemaps,
-          tileX: 0,  // tile X coordinate
-          tileY: 0,  // tile Y coordinate
-          tileZ: 1   // tile zoom level
-      }));
+      };
     }
 
-    this.leaflet.fitBounds(bounds);
+    this.map.fitWorld();
 
     Leaflet.control.scale({
       maxWidth: 250,
-    }).addTo(this.leaflet);
-    this.leaflet.attributionControl.setPrefix("");
+    }).addTo(this.map);
+    this.map.attributionControl.setPrefix("");
+
+    this.tileJsonRequests = [];
+
+    let tilesLoaded = 0;
+
+    tiles.forEach(tile => {
+      const { url, meta } = tile;
+
+      this.tileJsonRequests.push($.getJSON(url)
+          .done(info => {
+            const bounds = Leaflet.latLngBounds(
+                [info.bounds.slice(0, 2).reverse(), info.bounds.slice(2, 4).reverse()]
+              );
+            const layer = Leaflet.tileLayer(info.tiles[0], {
+                  bounds,
+                  minZoom: info.minzoom,
+                  maxZoom: info.maxzoom,
+                  tms: info.scheme === 'tms'
+                }).addTo(this.map);
+
+            // For some reason, getLatLng is not defined for tileLayer?
```

Comment:
I added that because calling layer.openPopup() calls getLatLng on the layer so it knows where to place the popup.